registry/data-models/common/models.py

```diff
@@ -0,0 +1,139 @@
+    RAGGED = "ragged"  # Ragged tensors (also known as nested tensors) are similar to dense tensors but have variable-length dimensions.
+
+
+class FeatureValueType(Enum):
+    """
+    The high level types associated with a feature.
+    This represents the high level semantic types supported by early versions of Frame.
+    """
+    BOOLEAN = "boolean"  # Boolean valued feature
+    NUMERIC = "numeric"  # Numerically valued feature
+    CATEGORICAL = "categorical"  # Represent a feature that consists of a single category
+    CATEGORICAL_SET = "categorical_set"  # Represent a feature that consists of multiple categories
+    DENSE_VECTOR = "dense_vector"  # Represent a feature in vector format where the majority of the elements are non-zero
+    TERM_VECTOR = "term_vector"  # Represent features that has string terms and numeric value
+    TENSOR = "tensor"  # Represent tensor based features.
+    UNSPECIFIED = "unspecified"  # Placeholder for when no types are specified
+
+
+class Dimension(BaseModel):
+    """
+    Tensor is used to represent feature data. A tensor is a generalization of vectors and matrices to potentially higher dimensions.
+    """
+    type: DimensionType  # Type of the dimension in the tensor. Each dimension can have a different type.
+    shape: Optional[int]  # Size of the dimension in the tensor. If unset, it means the size is unknown and actual size will be determined at runtime.
+
+
+class TensorFeatureFormat(BaseModel):
+    """
+    Defines the format of feature data. Feature data is produced by applying transformation on source, in a Feature.
+    Tensor is used to represent feature data. A tensor is a generalization of vectors and matrices to potentially
+    higher dimensions.
+    """
+    tensorCategory: TensorCategory  # Type of the tensor.
+    valueType: ValueType  # Type of the value column.
+    dimensions: List[Dimension]  # A feature data can have zero or more dimensions (columns that represent keys).
+
+
+class FeatureType(BaseModel):
+    """
+   Information about a featureName. It defines the type, format and default value.
+   Tensor is the next generation representation of the features, so using
+   Tensor type w TensorFeatureFormat would be preferable FeatureType.
+    """
+    type: FeatureValueType  # Defines the high level semantic type of feature.
+    format: Optional[TensorFeatureFormat]  # Defines the format of feature data.
+    defaultValue: Union[bool, int, float, str, bytes]
+
+
+class Clazz(BaseModel):
+    """
+    Reference to a class by fully-qualified name
+    """
+    fullyQualifiedName: str  # A fully-qualified class name including paths.
+
+
+class Function(BaseModel):
+    """
+    Base model for all functions
+    """
+    expression: str  # Expression in str format
+
+
+class MvelExpression(Function):
+    """
+    An expression in MVEL language.
+    """
+    mvel: str  # The MVEL expression
+
+
+class UserDefinedFunction(Function):
```

Comment:
Curious, how would `expression` be filled in this case?

Reply:
Since there is clazz name and parameters of UserDefinedFunction, we might need to add these two fields in experssion. I didn't implement this part as I am not sure how we want to show this in UI